appdev/connectors/mysql_connector.py

```diff
@@ -0,0 +1,123 @@
+from threading import Thread
+from Queue import Queue, Empty
+import mysql.connector
+
+
+class DbThread(Thread):
+  def __init__(self, connection_pool, table, input_queue, result_queue=None,
+               queue_timeout=0.5):
+    super(DbThread, self).__init__()
+    self.connection_pool = connection_pool
+    self.table = table
+    self.input_queue = input_queue
+    self.result_queue = result_queue
+    self.queue_timeout = queue_timeout
+
+
+class WriterThread(DbThread):
+  def run(self):
+    # Create query in the format:
+    #   "INSERT INTO salaries (emp_no, salary, from_date, to_date)
+    #     VALUES (%(emp_no)s, %(salary)s, %(from_date)s, %(to_date)s)"
+    # data is dictionary such as {'emp_no': 1, 'salary': 2, ...}
+    while not self.input_queue.empty():
+      connection = self.connection_pool.get()
+      cursor = connection.cursor()
+      try:
+        # A thread may enter the loop, but another thread could pluck the
+        # last item. This timeout is to prevent a thread from waiting for
+        # another item that will never come.
+        data = self.input_queue.get(timeout=self.queue_timeout)
+      except Empty:
+        break
+      values_string = ['%({})s'.format(col) for col in data.keys()]
```

Comment:
`values_array` - or join on `,` here.  Also, does `'%({})s'.format(col)` distinguish between data types in the columns?  For example, will it output `'lol'` rather than `lol` if my `col` variable is a string?  I couldn't find this syntax from a quick google search, but if you have a quick answer that would be great

Reply:
I believe it will cast `col` to a string no matter what data type it is, and the result of that is the same as if you were to do `print col` (meaning in your example it would output `lol`.

Reply:
Why do we wrap in `'%({})s'` in the first place

Reply:
Talked offline, looks good!